packages/flutter/lib/src/widgets/overscroll_indicator.dart

```diff
@@ -637,18 +644,285 @@ class _GlowingOverscrollIndicatorPainter extends CustomPainter {
+///  * [OverscrollIndicatorNotification], which can be used to prevent the stretch
+///    effect from being applied at all.
+///  * [NotificationListener], to listen for the
+///    [OverscrollIndicatorNotification]
+class StretchingOverscrollIndicator extends StatefulWidget {
+  /// Creates a visual indication that a scroll view has overscrolled by
+  /// applying a stretch transformation to the content.
+  ///
+  /// In order for this widget to display an overscroll indication, the [child]
+  /// widget must contain a widget that generates a [ScrollNotification], such
+  /// as a [ListView] or a [GridView].
+  ///
+  /// The [axisDirection] and [notificationPredicate] arguments must not be null.
+  const StretchingOverscrollIndicator({
+    Key? key,
+    required this.axisDirection,
+    this.notificationPredicate = defaultScrollNotificationPredicate,
+    this.child,
+  }) : assert(axisDirection != null),
+        assert(notificationPredicate != null),
+        super(key: key);
+
+  /// {@macro flutter.overscroll.axisDirection}
+  final AxisDirection axisDirection;
+
+  /// {@macro flutter.overscroll.axis}
+  Axis get axis => axisDirectionToAxis(axisDirection);
+
+  /// {@macro flutter.overscroll.notificationPredicate}
+  final ScrollNotificationPredicate notificationPredicate;
+
+  /// The widget below this widget in the tree.
+  ///
+  /// The overscroll indicator will apply a stretch effect to this child. This
+  /// child (and its subtree) should include a source of [ScrollNotification]
+  /// notifications.
+  ///
+  /// Typically a [StretchingOverscrollIndicator] is created by a
+  /// [ScrollBehavior.buildOverscrollIndicator] method when opted-in using the
+  /// [ScrollBehavior.updatedAndroidOverscrollIndicator] flag. In this case
+  /// the child is usually the one provided as an argument to that method.
+  final Widget? child;
+
+  @override
+  State<StretchingOverscrollIndicator> createState() => _StretchingOverscrollIndicatorState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(EnumProperty<AxisDirection>('axisDirection', axisDirection));
+  }
+}
+
+class _StretchingOverscrollIndicatorState extends State<StretchingOverscrollIndicator> with TickerProviderStateMixin {
+  _StretchController? _stretchController;
+  ScrollNotification? _lastNotification;
+  OverscrollNotification? _lastOverscrollNotification;
+  bool _accepted = true;
+
+  @override
+  void initState() {
+    super.initState();
+    _stretchController = _StretchController(vsync: this);
+  }
+
+  bool _handleScrollNotification(ScrollNotification notification) {
+    if (!widget.notificationPredicate(notification))
+      return false;
+
+    if (notification is OverscrollNotification) {
+      _lastOverscrollNotification = notification;
+      if (_lastNotification.runtimeType != OverscrollNotification) {
```

Comment:
This runtimeType check is strange. Why do we need this? Why are we disallowing subclasses of OverscrollNotification?

Reply:
Wouldn't subclasses be allowed? Since OverscrollNotificaton is the base class?
This is used the same way in the `GlowingOverscrollIndicator._handleScrollNotification` in order to manage dispatching the OverscrollIndicatorNotification - which lets folks call disallow on it.

Reply:
I think it is intentional, since we do the same in the glowing indicator. 
https://github.com/flutter/flutter/blob/4b538751b15340ef466f85d1249e7d2ee60de55b/packages/flutter/lib/src/widgets/overscroll_indicator.dart#L274
Git history was unhelpful. 
I can try your change for both and see if anything breaks, might make it easier to reason about and at least add a comment explaining.